### PR TITLE
Now based on ruby instead of bash.

### PR DESCRIPTION
**Pull request by hinnerk**

### CODE REVIEW

File: src/test/hostapd_test.rb [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+  def test_ht40_is_negative_when_channel_is_greater_than_7
+    8.upto 15 do |channel|
+      with_channel(channel) { Wifi.start }
+      assert_includes config, "ht_capab=[HT20][HT40-][SHORT-GI-20][SHORT-GI-40][DSSS_CCK-40][TX-STBC][RX-STBC1]"
+    end
+  end
+
+  ["private", "public"].each do |section|
+    define_method("test_#{section}_macaddr_acl_is_0") do
+      self.method("#{section}_disabled").call { Wifi.start }
```
Comment (colszowka) [inline src/test/hostapd_test.rb]: I`d suggest doing this in a slightly less meta way:

``` ruby
test "#{section}_macaddr_acl_is_0" do
   send("#{section}_disabled") { Wifi.start }
  # ....
end
```

Applies to all these dynamically defined methods. test "string for name" is a legitimate interface to define tests with dynamic names and send is the way to invoke methods by variable-sourced names, while define_method and retrieving method procs from objects I'd consider meta-programming tools. Imho that alternate version is a bit more readable too :)

/cc @boof

**Comment by colszowka** [comment]
@hinnerk Made a comment regarding the implementation of the tests, other than that looks reasonable :) I did not thoroughly check the actual functionality though, please let me know if I should do that as well

File: src/test/hostapd_test.rb (hunk 1 above)
Comment (hinnerk) [inline src/test/hostapd_test.rb]: Hmm: `in 'test': wrong number of arguments (given 1, expected 2) (ArgumentError)` @colszowka

**Comment by hinnerk** [comment]
Tis ain't no working yet:

```
Feb 09 17:40:24 sop01 systemd[1]: Started Run and configure the WIFI HOSTSPOT 'hostapd' service.
Feb 09 17:40:24 sop01 systemd-docker[12930]: D, [2016-02-09T17:40:24.654547 #1] DEBUG -- : fetching networks
Feb 09 17:40:24 sop01 systemd-docker[12930]: I, [2016-02-09T17:40:24.654619 #1]  INFO -- : found wifis: []
Feb 09 17:40:24 sop01 systemd-docker[12930]: D, [2016-02-09T17:40:24.654634 #1] DEBUG -- : fetching networks
Feb 09 17:40:24 sop01 systemd-docker[12930]: Configuration file: /etc/hostapd/hostapd.conf
Feb 09 17:40:24 sop01 systemd-docker[12930]: Could not open configuration file '/etc/hostapd/hostapd.conf' for reading.
Feb 09 17:40:24 sop01 systemd-docker[12930]: Failed to set up interface with /etc/hostapd/hostapd.conf
Feb 09 17:40:24 sop01 systemd-docker[12930]: Failed to initialize interface
```

/Cc @colszowka

**Pull request merged**